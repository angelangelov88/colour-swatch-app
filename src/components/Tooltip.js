import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'
import { InformationCircleIcon } from '@heroicons/react/outline'
import { useState } from 'react'

const Tooltip = (props) => {
  let {
    className,
    theme,
    selected,
    elementToClick,
    Icon,
    clickOnText,
    infoHeader,
    infoText,
    link,
    tooltipPosition,
    arrowPosition
  } = props


  const [clicked, setClicked] = useState(false)
  const handleClick = () => {
    if (selected === 'click') {
      setClicked(!clicked)
    }
  }

  const classes = classNames(
    "relative font-ibm w-fit text-gray-50 group",
    clicked && "text-gray-70",
    selected === "hover" && "hover:text-gray-70", 
    className
  )

  const tooltipClasses = classNames(
    clicked ? "flex" : "hidden",
    selected === "hover" && "group-hover:flex",
    theme === "light" ? "bg-white text-gray-80" : "bg-black text-white",
    "absolute w-28 h-28 flex-col justify-evenly items-center rounded-lg text-sm shadow-[0_10px_15px_-3px_rgba(0,0,0,0.1)] after:absolute after:content:none after:border-[10px] after:border-transparent",

    (tooltipPosition === "bottom" && arrowPosition === "centre") && "bottom-8 -left-11 -ml-0.5 after:left-[40.5%] after:-bottom-[8px] after:border-b-0 after:border-t-[white]",
    
    (tooltipPosition === "bottom" && arrowPosition === "centre" && theme === "dark") && "after:border-t-[black]",

    (tooltipPosition === "bottom" && arrowPosition === "left") && "bottom-8 -left-2 after:left-[7%] after:-bottom-[8px] after:border-b-0 after:border-t-[white]",

    (tooltipPosition === "bottom" && arrowPosition === "left" && theme === "dark") && "after:border-t-[black]",

    (tooltipPosition === "bottom" && arrowPosition === "right") && "bottom-8 -right-2 -ml-0.5 after:right-[7%] after:-bottom-[8px] after:border-b-0 after:border-t-[white]",

    (tooltipPosition === "bottom" && arrowPosition === "right" && theme === "dark") && "after:border-t-[black]",

    tooltipPosition === "top" && "top-8 -left-11 -ml-0.5 after:left-[41%] after:-top-4 after:border-b-[white] after:border-t-8",
    
    (tooltipPosition === "top" && theme === "dark") && "after:!border-b-[black]",
    
    tooltipPosition === "left" && "-top-12 left-8 after:-left-4 after:bottom-[40%] after:border-r-[white] after:border-t-10 after:border-l-8",

    (tooltipPosition === "left" && theme === "dark") && "after:border-r-[black] ",

    tooltipPosition === "right" && "-top-11 right-8 -mt-0.5 after:-right-5 after:bottom-[43%] after:border-b-8 after:border-l-[white]",

    (tooltipPosition === "right" && theme === "dark") && "after:border-l-[black] ",

    className
  )

  return ( 
    <div className={classes}>
      {elementToClick === "icon" ?
        <Icon 
          className="w-5 cursor-pointer"
          onClick={handleClick}
        />
        : 
        <div
          className="cursor-pointer"
          onClick={handleClick}
        >
          {clickOnText}
        </div>
      }
      <div className={tooltipClasses}>
        <h3 className="font-semibold">{infoHeader}</h3>
        <p>{infoText}</p>
        <a href={link} className="text-blue-60">Link</a>
      </div>
    </div>
  )
}

Tooltip.defaultProps = {
  theme: "light",
  selected: "hover",
  elementToClick: "icon",
  Icon: {InformationCircleIcon},
  clickOnText: "Please click here",
  infoHeader: "Info Header",
  infoText: "Info Text",
  link: "https://legalesign.com",
  tooltipPosition: "bottom",
  arrowPosition: "centre"
}

Tooltip.propTypes = {
  className: PropTypes.string,
  theme:PropTypes.oneOf([
    "dark",
    "light"
  ]),
  selected: PropTypes.oneOf([
    "hover",
    "click",
  ]),
  elementToClick: PropTypes.oneOf([
    "icon",
    "div"
  ]),
  Icon: PropTypes.object,
  clickOnText: PropTypes.string,
  infoHeader: PropTypes.string,
  infoText: PropTypes.string,
  link: PropTypes.string,
  tooltipPosition: PropTypes.oneOf([
    "bottom",
    "top",
    "left",
    "right",
  ]),
  arrowPosition: PropTypes.oneOf([
    "centre",
    "left",
    "right",
  ]),
}

export { Tooltip }
