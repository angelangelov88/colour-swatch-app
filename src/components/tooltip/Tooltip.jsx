import React from 'react'
import classNames from 'classnames'
import PropTypes from 'prop-types'

const Tooltip = (props) => {
  let {
    className,
    children,
    selected,
    Icon,
    clickOnText,
    tooltipPosition,
    arrowPosition
  } = props

  const [clicked, setClicked] = React.useState(false)

  const handleClick = () => {
    if (selected === 'click') {
      setClicked(!clicked)
    }
  }

  const classes = classNames(
    "relative font-ibm w-fit pt-2 text-gray-50 group",
    clicked && "text-gray-70",
    selected === "hover" && "hover:text-gray-70",
    className
  )

  const tooltipClasses = classNames(
    clicked ? "flex" : "hidden",
    selected === "hover" && "group-hover:flex",
    "bg-white dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center p-2",
    "absolute min-w-[112px] min-h-[112px] flex-col justify-evenly items-center rounded-lg shadow-lg",
    "after:absolute after:content:none after:border-[10px] after:border-transparent",
    (tooltipPosition === "bottom" && arrowPosition === "centre") && "bottom-8 -left-11 -ml-0.5 after:left-[40.5%] after:-bottom-[8px] after:border-b-0 after:border-t-white dark:after:border-t-gray-130",
    (tooltipPosition === "bottom" && arrowPosition === "left") && "bottom-8 -left-2 after:left-[7%] after:-bottom-[8px] after:border-b-0 after:border-t-white dark:after:border-t-gray-130",
    (tooltipPosition === "bottom" && arrowPosition === "right") && "bottom-8 -right-2 -ml-0.5 after:right-[7%] after:-bottom-[8px] after:border-b-0 after:border-t-white dark:after:border-t-gray-130",
    tooltipPosition === "top" && "top-8 -left-11 -ml-0.5 after:left-[41%] after:-top-4 after:border-b-white after:border-t-8 dark:after:border-b-gray-130",
    tooltipPosition === "left" && "-top-12 left-8 after:-left-4 after:bottom-[40%] after:border-r-white after:border-t-10 after:border-l-8 dark:after:border-r-gray-130",
    tooltipPosition === "right" && "-top-11 right-8 -mt-0.5 after:-right-5 after:bottom-[43%] after:border-b-8 after:border-l-white dark:after:border-l-gray-130",
    className
  )

  return (
    <div className={classes}>
      {Icon &&
        <Icon
          className="w-5 cursor-pointer"
          onClick={handleClick}
        />
      }
      {clickOnText &&
        <div
          className="cursor-pointer w-max"
          onClick={handleClick}
        >
          {clickOnText}
        </div>
      }
      <div className={tooltipClasses}>{children}</div>
    </div>
  )
}

Tooltip.defaultProps = {
  children: undefined,
  selected: "hover",
  tooltipPosition: "bottom",
  arrowPosition: "centre"
}

Tooltip.propTypes = {
  className: PropTypes.string,
  children: PropTypes.node,
  selected: PropTypes.oneOf(["hover", "click"]),
  Icon: PropTypes.object,
  clickOnText: PropTypes.string,
  tooltipPosition: PropTypes.oneOf([
    "bottom",
    "top",
    "left",
    "right",
  ]),
  arrowPosition: PropTypes.oneOf([
    "centre",
    "left",
    "right",
  ]),
}

export { Tooltip }
