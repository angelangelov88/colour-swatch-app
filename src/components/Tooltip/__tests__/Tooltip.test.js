import renderer from 'react-test-renderer'
import {Tooltip} from "../Tooltip"
import { InformationCircleIcon } from '@heroicons/react/outline'

it('should render with info icon', () => {
  const component = renderer.create(
  <Tooltip 
    Icon={InformationCircleIcon}
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
});

it('should render with info icon and under the icon', () => {
  const component = renderer.create(
  <Tooltip 
    Icon={InformationCircleIcon}
    tooltipPosition="top"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

it('should render with text instead of icon', () => {
  const component = renderer.create(
  <Tooltip 
    clickOnText="Click here"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render on click and with icon', () => {
  const component = renderer.create(
  <Tooltip 
    selected="click"
    Icon={InformationCircleIcon}
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})


it('should render with hover and text', () => {
  const component = renderer.create(
  <Tooltip 
    selected="hover"
    clickOnText="Click here"
  />,
  );
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})



