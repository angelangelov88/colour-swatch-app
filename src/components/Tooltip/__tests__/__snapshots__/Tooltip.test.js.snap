// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render on click and with icon 1`] = `
<div
  className="relative font-ibm w-fit pt-2 text-gray-50 group"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden bg-white dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center p-2 absolute min-w-[112px] min-h-[112px] flex-col justify-evenly items-center rounded-lg shadow-[0_10px_15px_-3px_rgba(0,0,0,0.1)] after:absolute after:content:none after:border-[10px] after:border-transparent bottom-8 -left-11 -ml-0.5 after:left-[40.5%] after:-bottom-[8px] after:border-b-0 after:border-t-white dark:after:border-t-gray-130"
  />
</div>
`;

exports[`should render with hover and text 1`] = `
<div
  className="relative font-ibm w-fit pt-2 text-gray-50 group hover:text-gray-70"
>
  <div
    className="cursor-pointer w-max"
    onClick={[Function]}
  >
    Click here
  </div>
  <div
    className="hidden group-hover:flex bg-white dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center p-2 absolute min-w-[112px] min-h-[112px] flex-col justify-evenly items-center rounded-lg shadow-[0_10px_15px_-3px_rgba(0,0,0,0.1)] after:absolute after:content:none after:border-[10px] after:border-transparent bottom-8 -left-11 -ml-0.5 after:left-[40.5%] after:-bottom-[8px] after:border-b-0 after:border-t-white dark:after:border-t-gray-130"
  />
</div>
`;

exports[`should render with info icon 1`] = `
<div
  className="relative font-ibm w-fit pt-2 text-gray-50 group hover:text-gray-70"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden group-hover:flex bg-white dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center p-2 absolute min-w-[112px] min-h-[112px] flex-col justify-evenly items-center rounded-lg shadow-[0_10px_15px_-3px_rgba(0,0,0,0.1)] after:absolute after:content:none after:border-[10px] after:border-transparent bottom-8 -left-11 -ml-0.5 after:left-[40.5%] after:-bottom-[8px] after:border-b-0 after:border-t-white dark:after:border-t-gray-130"
  />
</div>
`;

exports[`should render with info icon and under the icon 1`] = `
<div
  className="relative font-ibm w-fit pt-2 text-gray-50 group hover:text-gray-70"
>
  <svg
    aria-hidden="true"
    className="w-5 cursor-pointer"
    fill="none"
    onClick={[Function]}
    stroke="currentColor"
    strokeWidth={2}
    viewBox="0 0 24 24"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M13 16h-1v-4h-1m1-4h.01M21 12a9 9 0 11-18 0 9 9 0 0118 0z"
      strokeLinecap="round"
      strokeLinejoin="round"
    />
  </svg>
  <div
    className="hidden group-hover:flex bg-white dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center p-2 absolute min-w-[112px] min-h-[112px] flex-col justify-evenly items-center rounded-lg shadow-[0_10px_15px_-3px_rgba(0,0,0,0.1)] after:absolute after:content:none after:border-[10px] after:border-transparent top-8 -left-11 -ml-0.5 after:left-[41%] after:-top-4 after:border-b-white after:border-t-8 dark:after:border-b-gray-130"
  />
</div>
`;

exports[`should render with text instead of icon 1`] = `
<div
  className="relative font-ibm w-fit pt-2 text-gray-50 group hover:text-gray-70"
>
  <div
    className="cursor-pointer w-max"
    onClick={[Function]}
  >
    Click here
  </div>
  <div
    className="hidden group-hover:flex bg-white dark:bg-gray-130 text-gray-80 dark:text-gray-10 text-sm text-center p-2 absolute min-w-[112px] min-h-[112px] flex-col justify-evenly items-center rounded-lg shadow-[0_10px_15px_-3px_rgba(0,0,0,0.1)] after:absolute after:content:none after:border-[10px] after:border-transparent bottom-8 -left-11 -ml-0.5 after:left-[40.5%] after:-bottom-[8px] after:border-b-0 after:border-t-white dark:after:border-t-gray-130"
  />
</div>
`;
